pkg/utils/kubernetes/owners.go

```diff
@@ -38,6 +42,46 @@ func SetOwnerForObject(obj, owner client.Object) {
 	}
 }
 
+// managingObjectT is an interface that is used to represent a managing object.
+// managingObjects can be of type Gateway, ControlPlane, or DataPlane.
```

Comment:
Nit: While this uses the `interface` keyword this is a [type constraint](https://go.dev/blog/intro-generics#type-sets).

I'd reword this as:

```suggestion
// managingObjectT is type constraint that is used to represent a managing object.
// Currently it can be one of: Gateway, ControlPlane, or DataPlane.
```